Sources/LanguageServerProtocol/SupportTypes/RegistrationOptions.swift

```diff
@@ -0,0 +1,55 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+/// FIXME: Find a way to abstract this without deserializing.
```

Comment:
I'm not sure of a better generalization here, we could do something like what we do for Notifications/Requests but to store it we'd need to type erase it/box it (e.g. an AnyRegistrationOptions)

Reply:
There aren't that many of these - is there a reason not to just make `encodeToLSPAny` a protocol requirement and implement it concretely for each concrete type that implements the protocol?